src/FizzWare.NBuilder/ValueSet.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace FizzWare.NBuilder.Tests.Unit
+{
+    public class ValueSet<T>: IEnumerable<T>

```

Comment:
I'm curious why you're inheriting from `IEnumerable<T>` ? Like, I understand why you want a custom collection .. but what is the real customization here? `Random()` ? If that's it, why not make a simple extension method in `IEnumerable<T>` ?

Reply:
`ValueSet` is a collection construct--it's just one with specialized behavior. The optimization is `Next()` which loops around when it reaches the end of the collection, starting over from the beginning.